### PR TITLE
performance: introduce automated performance dashboard with pyroscope

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
### 1. Explain what the PR does

Tracee developers are struggling to find different ways to check performance results when doing the development. This is an attempt to automate performance checks and dashboards in order for an unified experience.

This first feature adds a grafana dash board containing trace pperf 

commit bb367ac4 (HEAD -> new-pyro, rafaeldtinoco/new-pyro)
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Mon Apr 3 06:09:03 2023

    performance: automate pyroscope dashboard view

    Developer might execute:

    - make -f builder/Makefile.performance pyroscope-start
    - make -f builder/Makefile.performance pyroscope-stop

    And will get access to a performance dashboard while executing tracee.

commit b3382a55
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Mon Apr 3 06:08:22 2023

    performance: create a pyroscope pperf client feature

### 2. Explain how to test it

```
$ make -f builder/Makefile.performance pyroscope-start

<open browser at http://localhost:3000>
```

image

> You can now navigate the dashboard, picking specific timeslices that will contain not only the event and CPU metrics, but also the flamegraph for that time window.

```
$ make -f builder/Makefile.performance pyroscope-start
```

### 3. Other comments

Idea is to make the dashboard better with time:

- include per application CPU and MEM consumption
- include 1 graph per eBPF program overhead
- ...

### CODE REVIEW

File: cmd/tracee-rules/main.go
Comment (NDStrahilevitz) [inline cmd/tracee-rules/main.go:162]: Missing the flag declaration in this file.

**Review by NDStrahilevitz — COMMENTED** [review]
Awesome work. There's a missing flag declaration and a question I have before I +1 but overall LGTM.

File: pkg/server/server.go
Comment (rafaeldtinoco) [reply pkg/server/server.go:92]: We already covered that in private. More performatic and I also couldn't do the "pull mode" (which is what you are referring) to work in an automated way: https://pyroscope.io/docs/golang-pull-mode/. 

I'll give another try, but not a blocker as we've spoken.

**Comment by NDStrahilevitz** [comment]
@josedonizetti waiting for you before merging or can we go ahead?

**Comment by rafaeldtinoco** [comment]
Adding to milestone 0.13.1 as this is not a code change and allows us to test Tracee better.

**Review by josedonizetti — APPROVED** [review]
NICE!

**Pull request merged**
